test/tools/unified-spec-runner/match.ts

```diff
@@ -497,12 +502,22 @@ function compareEvents(
         entities,
         path
       );
+      if (expectedEvent.commandSucceededEvent.hasServerConnectionId) {
+        expect(actualEvent).property('serverConnectionId').to.exist;
+      } else if (expectedEvent.commandSucceededEvent.hasServerConnectionId === false) {
+        expect(actualEvent).property('serverConnectionId').to.not.exist;
+      }
     } else if (expectedEvent.commandFailedEvent) {
       const path = `${rootPrefix}.commandFailedEvent`;
       if (!(actualEvent instanceof CommandFailedEvent)) {
         expect.fail(`expected ${path} to be instanceof CommandFailedEvent`);
       }
       compareCommandFailedEvents(actualEvent, expectedEvent.commandFailedEvent, entities, path);
+      if (expectedEvent.commandFailedEvent.hasServerConnectionId) {
+        expect(actualEvent).property('serverConnectionId').to.exist;
+      } else if (expectedEvent.commandFailedEvent.hasServerConnectionId === false) {
+        expect(actualEvent).property('serverConnectionId').to.not.exist;
```

Comment:
```suggestion
        expect(actualEvent).property('serverConnectionId').to.be.null;
```

Now that we are making it default to null, might as well test for that explicitly to catch any mistakes.